src/ristretto/dalek_range_proof.rs

```diff
@@ -88,6 +90,60 @@ impl RangeProofService for DalekRangeProofService {
     fn range(&self) -> usize {
         self.range
     }
+
+    fn construct_proof_with_rewind_key(&self, key: &RistrettoSecretKey, value: u64, rewind_key: &RistrettoSecretKey, rewind_blinding_key: &RistrettoSecretKey, proof_message: &[u8; 19]) -> Result<Vec<u8>, RangeProofError> {
+        let mut pt = Transcript::new(b"tari");
+        let mut full_proof_message = [0u8; 23];
+        full_proof_message[0..4].clone_from_slice(b"tari");
```

Comment:
Extract message as a `const`; And I'd highly recommend making it 2 bytes to give a precious extra 2 bytes to the message. 1 byte is a bit short 1:256 chance of getting a false positive, 2 bytes is 1:65k.

Reply:
Will do, was just following the precedent set by the transcript but I agree.

The select of how many bytes to use felt tricky. Happy to make it 2 for a 1 out of 65k chance of a false positive. Is that truly big enough? Maybe split the difference for a 3 byte check?